src/table/use-table-focus-navigation.ts

```diff
@@ -64,11 +62,14 @@ function useTableFocusNavigation<T extends { editConfig?: TableProps.EditConfig<
 
   const shiftFocus = useCallback(
     (vertical: -1 | 0 | 1, horizontal: -1 | 0 | 1) => {
-      // istanbul ignore if next
-      if (!currentFocusCell.current) {
+      const focusedCell = tableRoot.current?.querySelector<HTMLTableCellElement>('td:focus-within');
```

Comment:
The focus navigation partially broke here because of the focus events. I decided to reimplement this using proper CSS selectors instead of dynamically assigning focus handlers. Much shorter, too :)